From 812573e0f5cd77af712941a494409137631a7585 Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Mon, 05 Aug 2024 17:28:54 +0300
Subject: [PATCH] [SW-194151] g3: flash: override default prot to cover only 128kB

Default PROT_MASK protects all the four flash blocks by default
in the common spi flash driver in ppboot. This patch overrides
the default PROT_MASK for gaudi3 to protect only 128KB of the
flash space where recovery image resides.

Change-Id: If402e6977215454dca6a5bf7d951193a37fa9a90
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/arch/gaudi3/arc/common.h b/arch/gaudi3/arc/common.h
index 7b7a17c..e5d990a 100644
--- a/arch/gaudi3/arc/common.h
+++ b/arch/gaudi3/arc/common.h
@@ -126,4 +126,11 @@
  */
 #define SPI_RX_POLL_TIMEOUT_US		2560 /* timeout in us */
 
+/* Let this file be included in any order and override the PROT_MASK */
+#ifdef PROT_MASK
+#undef PROT_MASK
+#endif
+/* Only BP1 == 0x2 - means 2 blocks - up to 128KB (eROM) */
+#define PROT_MASK	(READ_STATUS_BP1 | READ_STATUS_SRWD)
+
 #endif /* COMMON_GAUDI3_ARC_GNU_H_ */
